Numeric Suite: Swap with shadow modules only for quantized part of model

raghuramank100 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#51052 Numeric Suite: Swap with shadow modules only for quantized part of model**

Ensure that shadow modules are inserted only for quantized modules in a model. Removes redundant module insertion.

Differential Revision: [D26054016](https://our.internmc.facebook.com/intern/diff/D26054016/)

raghuramank10000 committed:
Numeric Suite support

Ensure that shadow modules are inserted only for quantized modules in a model. Removes redundant module insertion.

Differential Revision: [D26054016](https://our.internmc.facebook.com/intern/diff/D26054016/)

[ghstack-poisoned]

raghuramank10000 committed:
Update on "Numeric Suite support"


Ensure that shadow modules are inserted only for quantized modules in a model. Removes redundant module insertion.

Differential Revision: [D26054016](https://our.internmc.facebook.com/intern/diff/D26054016/)

[ghstack-poisoned]

vkuzo commented on torch/quantization/_numeric_suite.py:
```diff
@@ -314,7 +315,7 @@ def prepare_model_with_stubs(float_module, q_module, module_swap_list, Logger):
             prepare_model_with_stubs(float_mod, mod, module_swap_list, Logger)
 
-        if type(float_mod) in module_swap_list:
+        if type(float_mod) in module_swap_list and type(mod) is not type(float_mod):
```
nit: can we add a comment to the code describing why?

raghuramank10000 committed:
Update on "Numeric Suite: Swap with shadow modules only for quantized part of model"


Ensure that shadow modules are inserted only for quantized modules in a model. Removes redundant module insertion.

Differential Revision: [D26054016](https://our.internmc.facebook.com/intern/diff/D26054016/)

[ghstack-poisoned]

raghuramank10000 committed:
Update on "Numeric Suite: Swap with shadow modules only for quantized part of model"


Ensure that shadow modules are inserted only for quantized modules in a model. Removes redundant module insertion.

Differential Revision: [D26054016](https://our.internmc.facebook.com/intern/diff/D26054016/)

[ghstack-poisoned]

raghuramank10000 committed:
Update on "Numeric Suite: Swap with shadow modules only for quantized part of model"


Ensure that shadow modules are inserted only for quantized modules in a model. Removes redundant module insertion.

Differential Revision: [D26054016](https://our.internmc.facebook.com/intern/diff/D26054016/)

[ghstack-poisoned]

raghuramank10000 committed:
Update on "Numeric Suite: Swap with shadow modules only for quantized part of model"


Ensure that shadow modules are inserted only for quantized modules in a model. Removes redundant module insertion.

Differential Revision: [D26054016](https://our.internmc.facebook.com/intern/diff/D26054016/)

[ghstack-poisoned]

raghuramank10000 committed:
Update on "Numeric Suite: Swap with shadow modules only for quantized part of model"


Ensure that shadow modules are inserted only for quantized modules in a model. Removes redundant module insertion.

Differential Revision: [D26054016](https://our.internmc.facebook.com/intern/diff/D26054016/)

[ghstack-poisoned]
